Android/CameraSample/app/src/main/java/com/bentley/sample/camera/ImageCache.kt

```diff
@@ -0,0 +1,114 @@
+/*---------------------------------------------------------------------------------------------
+* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+* See LICENSE.md in the project root for license terms and full copyright notice.
+*--------------------------------------------------------------------------------------------*/
+package com.bentley.sample.camera
+
+import android.content.Intent
+import android.net.Uri
+import android.webkit.WebResourceResponse
+import androidx.core.content.FileProvider
+import com.bentley.sample.shared.FileHelper
+import com.bentley.sample.shared.MainActivity
+import com.eclipsesource.json.Json
+import com.eclipsesource.json.JsonValue
+import java.io.File
+import java.io.FileInputStream
+import java.time.LocalDateTime
+import java.time.format.DateTimeFormatter
+
+object ImageCache {
+    private const val urlScheme = "com.bentley.itms-image-cache"
+
+    private val baseDir: String by lazy {
+        CameraITMApplication.appContext.getExternalFilesDir("images").toString()
+    }
+
+    fun getDestinationDir(input: JsonValue?): String {
+        val iModelId = input?.asObject()?.get("iModelId")?.asString() ?: "unknownModelId"
```

Comment:
I think this should call getIModelId:
` val iModelId = getIModelId ?: "unknownModelId"`

Reply:
Fixed